File: 02_selection_sort/Python/01_selection_sort.py
def find_Smallest(arr):
    smallest = arr[0] # Для хранения наименьшего значения
    smallest_index = 0 # Для хранения индекса наименьшего значения
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index

def selection_Sort(arr):
    newArr = []
    for i in range(len(arr)):
        smallest = find_Smallest(arr) # находим наименьший элемент в массиве
        newArr.append(arr.pop(smallest)) # и добавляем его в новый массив
    return newArr

print(selection_Sort([-4, 5, 3, -2, 6, 2, 10, -1]))

File: 02_selection_sort/JS/01_selection_sort.js
"use strict";

const findSmallest = (arr) => {
    let smallest = arr[0];
    let smallestIndex = 0;

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallestIndex = i;
        }
    }
    return smallestIndex;
};

const selectionSort = (arr) => {
    const newArr = [];
    const length = arr.length;
    for (let i = 0; i < length; i++) {
        const smallest = findSmallest(arr);
        newArr.push(arr.splice(smallest, 1)[0]);

    }
    return newArr;
}

console.log(selectionSort([-4, 5, 3, -2, 6, 2, 10, -5]));

function newFunction(arr, smallest) {
    console.log(arr.splice(smallest, 1)[0]);
}
